File: is_prime/is_prime.cpp
#include <cmath>
#include <iostream>
#include <ostream>
#include <vector>
#include <ctime>
using namespace std;

// input: integer >=0
// output: find the factorial of ^ integer
bool is_prime_linear(int num)
{
  if (num <= 1)
    return false;
  for(int i = 2; i <= int(num/2); i++)
    if(num % i == 0)
      return false;

  return true;
}

bool is_prime_sqrt(int num)
{
  if (num <= 1)
    return false;
  int sq_n = int(sqrt(num));

  for(int i = 2; i <= sq_n; i++)
  {
    if(num % i == 0)
      return false;
  }
  return true;
}

// To execute C++, please define "int main()"
int main() {
  int n = 111781;
  // int n = 131;
  std::clock_t start = std::clock();
  cout<<CLOCKS_PER_SEC<<endl;
  if(is_prime_sqrt(n))
    cout << "The number " << n << " is prime" << "\n";
  else
    cout << "The number " << n << " is not prime" << "\n";  
  std::clock_t end = std::clock();
  double elapsed_time = double(end - start) / CLOCKS_PER_SEC;

  cout<< "Time taken to check if a number is prime: " << elapsed_time << " seconds" << "\n";
  return 0;
}

File: is_prime/is_prime.py
import numpy as np
import time

def is_prime_linear(num):
    if num <= 1:
        return False

    for i in range(2, (num//2)+1):
        if num % i == 0:
            return False

    return True

def is_prime_sqrt(num):
    if num <= 1:
        return False
    n_sq = int(np.sqrt(num))
    for i in range(2, n_sq+1):
        if num % i == 0:
            return False
    return True


if __name__ == "__main__":
    n = 111781
    start = time.time()
    if is_prime_sqrt(n):
        print("The number", n, "is prime")
    else:
        print("The number", n, "is not prime")

    end = time.time()
    print("Time taken is:", round(end-start, 6), "seconds")
